### PR TITLE
microsocks: add low resource SOCKS5 TCP/IP only proxy server

### PR DESCRIPTION
**Pull request by matkor**
Initial version 1.0.3

Signed-off-by: Mateusz Korniak <matkorgithubcom@ant.gliwice.pl>

Maintainer: me
Compile tested: mipsel_24kc, Ubuntu
Run tested: mipsel_24kc , Netgear WAC104, OpenWrt 21.02.3 r16554-1d4dea6d4f / LuCI openwrt-21.02 branch git-22.083.69138-0a0ce2a
Tested on production load for few weeks now, without issues.

Description:
Tiny TCP/IP only, SOCKS5 only proxy server.

### CODE REVIEW

File: net/microsocks/Makefile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+PKG_SOURCE_URL:=http://ftp.barfooze.de/pub/sabotage/tarballs/
+PKG_HASH:=02449160c505ff7ea1bcb71ae781dafcfe66ab95287f3a4fca11621b74b2d7ac
+PKG_LICENSE:=MIT
+PKG_MAINTAINER:=Mateusz Korniak <matkorgithubcom@ant.gliwice.pl>
+
+include $(INCLUDE_DIR)/package.mk
+
+# Package definition; instructs on how and where our package will appear in the overall configuration menu ('make menuconfig')
+define Package/microsocks
+  SECTION:=net
```
Comment (diizzyy) [inline net/microsocks/Makefile:16]: You probably want to add
```SUBMENU:=Web Servers/Proxies```

File: net/microsocks/Makefile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  CATEGORY:=Network
+  TITLE:=SOCKS5 TCP/IP only proxy
+  DEPENDS:=+libc +libpthread
```
Comment (diizzyy) [inline net/microsocks/Makefile:19]: Are you sure these are needed to be defined?

File: net/microsocks/Makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+define Package/microsocks/conffiles
+/etc/config/microsocks
+endef
+
+# define Build/Compile
+# 	# $(MAKE) clean
+# 	$(MAKE) -C $(PKG_BUILD_DIR) CC="$(TARGET_CC)" CFLAGS="$(TARGET_CFLAGS)" LDFLAGS="$(TARGET_LDFLAGS)"	
+# endef
+
```
Comment (diizzyy) [inline net/microsocks/Makefile:35]: Not sure if it makes a difference but when I ported it why back these settings helped a bit:
```
TARGET_CFLAGS+= -flto
TARGET_LDFLAGS+= -flto -Wl,--gc-sections,--as-needed
```

File: net/microsocks/Makefile (hunk 1 above)
Comment (matkor) [reply net/microsocks/Makefile:15]: Right. Added.

File: net/microsocks/Makefile (hunk 2 above)
Comment (matkor) [reply net/microsocks/Makefile]: Seems be not needed. Pruned.

File: net/microsocks/Makefile (hunk 3 above)
Comment (matkor) [reply net/microsocks/Makefile:36]: Reduces bin size by few bytes. Added.

**Comment by matkor** [comment]
Can I just push commits trying to solve mentioned issues for now, and when we are (hopefully) done, squash them to one for merge?

**Comment by diizzyy** [comment]
Sure, go ahead

**Comment by matkor** [comment]
> Sure, go ahead

So I pushed all fixes here.
Please let me know, when I should squash all commits  again, or what to do now :) . TIA

**Comment by diizzyy** [comment]
Squash commits and you're good to go

**Comment by matkor** [comment]
Reopened as
https://github.com/openwrt/packages/pull/19276
as I broke branch history for this PR

**Pull request closed without merging**